dali/operators/image/convolution/gaussian_blur.cc

```diff
@@ -0,0 +1,287 @@
+    DALI_ENFORCE(tensor.shape()[0] == 1 || tensor.shape()[0] == argument_length,
+                 make_string("Argument ", name, " for sample ", sample_idx,
+                             " is expected to have shape equal to {1} or {", argument_length,
+                             "}, got: ", tensor.shape(), "."));
+    if (tensor.shape()[0] == 1) {
+      for (int i = 0; i < argument_length; i++) {
+        result[i] = tensor.data<T>()[0];
+      }
+    } else {
+      memcpy(result.data(), tensor.data<T>(), sizeof(T) * argument_length);
+    }
+    return;
+  }
+  std::vector<T> tmp;
+  // we already handled the argument input, this handles spec-related arguments only
+  GetSingleOrRepeatedArg(spec, tmp, name, argument_length);
+  memcpy(result.data(), tmp.data(), sizeof(T) * argument_length);
+}
+
+template <int axes>
+GaussianSampleParams<axes> GetSampleParams(int sample, const OpSpec& spec,
+                                           const ArgumentWorkspace& ws) {
+  GaussianSampleParams<axes> params;
+  GetGeneralizedArg<float>(make_span(params.sigmas), kSigmaArgName, sample, spec, ws);
+  GetGeneralizedArg<int>(make_span(params.window_sizes), kWindowSizeArgName, sample, spec, ws);
+  for (int i = 0; i < axes; i++) {
+    DALI_ENFORCE(
+        !(params.sigmas[i] == 0 && params.window_sizes[i] == 0),
+        make_string("`sigma` and `window_size` shouldn't be 0 at the same time for sample: ",
+                    sample, ", axis: ", i, "."));
+    DALI_ENFORCE(params.sigmas[i] >= 0,
+                 make_string("`sigma` must have non-negative values, got ", params.sigmas[i],
+                             " for sample: ", sample, ", axis: ", i, "."));
+    DALI_ENFORCE(params.window_sizes[i] >= 0,
+                 make_string("`window_size` must have non-negative values, got ",
+                             params.window_sizes[i], " for sample: ", sample, ", axis : ", i, "."));
+    if (params.window_sizes[i] == 0 && params.sigmas[i] > 0.f) {
+      params.window_sizes[i] = GaussianSigmaToDiameter(params.sigmas[i]);
+    } else if (params.sigmas[i] == 0.f && params.window_sizes[i] > 0) {
+      params.sigmas[i] = GaussianDiameterToSigma(params.window_sizes[i]);
+    }
+  }
+  return params;
+}
+
+GaussianDimDesc ParseAndValidateDim(int ndim, TensorLayout layout) {
+  static constexpr int kMaxDim = 3;
+  if (layout.empty()) {
+    // assuming plain data with no channels
+    DALI_ENFORCE(ndim <= kMaxDim,
+                 make_string("Input data with empty layout cannot have more than ", kMaxDim,
+                             " dimensions, got input with ", ndim, " dimensions."));
+    return {0, ndim, false, false};
+  }
+  // not-empty layout
+  int axes_start = 0;
+  int axes_count = ndim;
+  bool has_channels = ImageLayoutInfo::HasChannel(layout);
+  if (has_channels) {
+    axes_count--;
+    DALI_ENFORCE(ImageLayoutInfo::IsChannelLast(layout),
+                 "Only input data with no channels or channel-last is supported.");
```

Comment:
There's nothing that stops us from supporting channel last - just treat it as a sequence.

Reply:
Done